### PR TITLE
Whitelist query pagination

### PR DESCRIPTION
**Pull request by NickGonzalez04**
### What changes were made?
- pagination implementation against Whitelist data.
- initial data to be sent back as 'pending' if an admin user.
- removed `useWhitelist` hook since it was combined with `useWhitelistFilter` hook.
- backend endpoint to send back more data items if there is an included 'LastKeyEvaluated'
- formatted  `fetcher` and `getDynamoDBClient()` into a 'libs/' folder to use across the code base / endpoints.

### Is there any background info about the change?

- We needed to limit how much whitelist data was being to sent to frontend. This list will grow larger over time, so this does prevent any future rendering issues. 

Future Improvement Ideas:  
- Can be expanded to filter against recent dates
- Other attributes using conditional check with 'filterExpressions' in the backend.

### Are there any state changes? Mention key ones (if any)

- Just changing parameter value in `useWhiteListFilter` hook to include query and user wallet address.
- Tracking any state for filtered whitelist data.

### Screenshots or Gifs✨

https://user-images.githubusercontent.com/48919838/185991129-3128651a-ccc9-4412-a677-afee293dafb0.mp4

### Any new dependencies? Why were they added?

N/A

### Shortcut Link

https://app.shortcut.com/twali/story/1258/update-frontend-whitelist-to-fetch-on-scroll 
&
https://app.shortcut.com/twali/story/1254/update-getwhitelist-endpoint-to-paginate-results

### CODE REVIEW

File: data/dynamodb.js [hunk 1]
```diff
@@ -618,27 +621,62 @@ module.exports = {
     return dbUser;
   },
 
-  getWhitelist: async (userId) => {
+// Will most likely better serve us to set `whitelistStatus` as a secondary Index.
```
Comment (nagmak) [inline data/dynamodb.js:624]: Yeah I agree, that's a really good point

**Review by nagmak — APPROVED** [review]
Great stuff! Looks fireee.

**Comment by NickGonzalez04** [comment]
> 

hmm good point good point. Let me move it over into utils to not cause confusion! @nagmak

**Comment by nagmak** [comment]
Appreciate it!! @NickGonzalez04

**Pull request merged**